Sources/Purchasing/StoreKit2/StoreKit2PurchaseIntentListener.swift

```diff
@@ -139,6 +139,39 @@ struct StorePurchaseIntent: Sendable, Equatable {
     @available(tvOS, unavailable)
     @available(watchOS, unavailable)
     @available(visionOS, unavailable)
-    let purchaseIntent: PurchaseIntent?
+    let purchaseIntent: (any StoreKit2PurchaseIntentType)?
     #endif
+
+    static func == (lhs: StorePurchaseIntent, rhs: StorePurchaseIntent) -> Bool {
+    #if compiler(>=5.10)
+        return lhs.purchaseIntent?.id == rhs.purchaseIntent?.id
+    #else
+        return true
+    #endif
+    }
+}
+
+@available(iOS 16.4, macOS 14.4, *)
+@available(tvOS, unavailable)
+@available(watchOS, unavailable)
+@available(visionOS, unavailable)
+protocol StoreKit2PurchaseIntentType: Equatable, Identifiable, Sendable {
```

Comment:
I introduced this protocol to be able to test that the delegate call in `PurchasesOrchestrator.swift` does indeed call the track diagnostics method. Another option would have been to use a wrapper with a generic, but I didn't want to overcomplicate things so much only to have type safety in unit tests